### DISCUSSION TITLE
The Projection Storer does not handle the Debezium tombstone event

### DISCUSSION THREAD

**Discussion by lucascanna** [mia-platform/community#497] [Q&A] [answered]
### What product are you having troubles with?

Fast Data

### What Console version are you using?

12.4.0

### Description

Hi guys!

In order to allow Log Compaction on the ingestion topics, by default, Debezium’s Oracle connector always follows a delete event with a special tombstone event that has null value (as explained [here](https://debezium.io/documentation/reference/2.5/connectors/oracle.html#oracle-delete-events)).

Appearently, when the projection-storer reads this tombstone event, it fails with the following log message:
```
ERROR:
    @timestamp: "2024-05-16T10:50:21.176329079Z"
    message: "error writing projection records into storage"
    logger_name: "eu.miaplatform.fastdata.ps.app.storage.StorageVerticle"
    thread_name: "DefaultDispatcher-worker-3"
    code: "FD_PS_E4001"
    explanation: "empty projections operations list provided to the storage component"
    cause: null
    documentation: "https://docs.mia-platform.eu/docs/fast_data/troubleshooting/fast_data_error_codes#projection-storer"
    details: {
      "projection": "MY_TABLE",
      "batchID": "kafka_23_1715856621089"
    }
```
To solve this issue, I need to disable the property `tombstones.on.delete` on Debezium configuration but this prevents me from using log compaction on the ingestion topics.

Is there a way to avoid this situation?

Thanks!

**Comment by danibix95** [comment]
Hi, `projection-storer` supports `tombstone` events. The problem arise when the whole  batch of ingestion messages contains only events whose payload is empty. 
In such case, the internal logic treat this situation as a processing error, to prevent reaching the storage layer when not needed. This error is described in the [error codes page](https://docs.mia-platform.eu/docs/fast_data/troubleshooting/fast_data_error_codes#projection-storer).  

We may retain the logic to save external requests, but treat the situation as a warning.

**Reply by lucascanna** [reply]
Thanks @danibix95!

For the moment I will disable the tombstone events anyway to avoid this problem. I will wait for the fix you have suggested to enable tombstones again.

**Reply by danibix95** [reply]
Hello, with version `v1.2.2` of projection storer (which will be released with next Console version), this situation won't arise anymore and you'll be able to use `tombstone` events without any further trouble.

**Reply by lucascanna** [reply, marked as answer]
I confirm this is solved with v1.2.2
